src/lib/common/file/file.directive.ts

```diff
@@ -64,4 +67,18 @@ export class FileDirective extends NgComponent {
 			this.value = e.target.result;
 		}
 	}
+
+	hideBackdrop() {
+		if (this.backdropService) {
+			this.backdropService.hide();
+		}
+	}
+
+	revealBackdrop() {
+		if (this.backdropService) {
+			if (!this.backdropService.isVisible) {
```

Comment:
you don't need to worry about that.

Reply:
but we have to)
we should either check this state (!this.backdropService.isVisible) and place setTimeout here, in file.directive.ts or
we can get rid of this checking here, but place setTimeout in BackdropService file.
otherwise, it won't work correctly due to invalid calling sequence.
or i just do it wrong)

Reply:
ok, rename it to isActive please